Tools/yocto/README.md

```diff
@@ -0,0 +1,258 @@
+```
+
+There you can run ```bitbake``` commands for the generated Yocto project.
+
+### Speeding the build with an external Yocto cache
+
+Rebuilding the whole toolchain and image from scratch is slow, and if you
+are developing with this you will end having to rebuild the toolchains and
+images from scratch usually.
+
+A trick to speed up the whole rebuild process is to store the yocto sstate
+and download caches outside of the WebKitBuild directory, so they can be
+reused between different builds.
+
+To enable that, add this lines to your ```~/.bashrc``` file
+
+```
+# Export shared directories for yocto caches
+export DL_DIR="${HOME}/.cache/yocto/downloads"
+export SSTATE_DIR="${HOME}/.cache/yocto/sstate"
+export BB_ENV_PASSTHROUGH_ADDITIONS="${BB_ENV_PASSTHROUGH_ADDITIONS} DL_DIR SSTATE_DIR"
+```
+
+### Disabling automatic wipe on change
+
+The script ```cross-toolchain-helper``` will automatically wipe and rebuild the
+cross-toolchain and images for the target board if it detects anything has changed
+on the target source definitions.
+
+To avoid this you can pass the flag ```--no-wipe-on-change``` to ```cross-toolchain-helper```
+or you can set the environment variable ```WEBKIT_CROSS_WIPE_ON_CHANGE=0```
+
+However it is not recommended that you do this, instead try to enable an external Yocto cache
+to have very fast rebuilds (see above).
+
+
+### Disabling export default environment
+
+The script ```cross-toolchain-helper``` will automatically export several environment variables
+defined in file ```${WebKitDirectory}/Tools/yocto/targets.conf``` for the given target.
+This environment variables are usually used to change the default build parameters for the
+script ```build-webkit```
+
+If you want to disable this you can pass the flag ```--no-export-default-environment``` to ```cross-toolchain-helper```
+or you can set the environment variable ```WEBKIT_CROSS_EXPORT_ENV=0```
+
+
+## Practical examples on how to use this
+
+This is an example on how to build and run WPE for development on the RPi:
+
+### Build the base image, flash the board and clone the WebKit repo
+
+1. Build the image that will be flashed on the board:
+
+```
+user@workstation $ Tools/Scripts/cross-toolchain-helper --cross-target=rpi3-32bits-mesa --build-image
+```
+
+2. At the end of the build it will print the path to the image, if you missed it
+   run the command again and it will print the path again (doesn't rebuild if not needed).
+   Example:
+
+```
+user@workstation $ Tools/Scripts/cross-toolchain-helper --cross-target=rpi3-32bits-mesa --build-image
+cross-toolchain-helper INFO: Image already built at: /home/igalia/webkit/WebKitBuild/CrossToolChains/rpi3-32bits-mesa/build/image/core-image-webkit-dev-ci-tools-tests.wic.bz2
+```
+
+3. Flash a SDCard with this image
+
+```
+user@workstation $ cat /path/to/core-image-webkit-dev-ci-tools-tests.wic.bz2 | bzip2 -dc | dd of=/dev/mmcblk0 status=progress bs=4k
```

Comment:
This is not very fast/efficient. Maybe mention `bmaptool`, there's a nice blog post about it, https://blog.lazy-evaluation.net/posts/linux/bmaptool.html

Reply:
Good tip. I will add info about bmaptool to the README. Thanks